### ISSUE TITLE
Support non 'inherit rotation' bone

### ISSUE THREAD

**Issue by Jason0214** [godotengine/godot-blender-exporter#140]
**Issue description:**

currently export armature animation with 'inherit rotation' unchecked bones would generate a different animation in Godot, as Godot treat bones all inherit from parent.
<img width="390" alt="screen shot 2019-01-18 at 09 01 40" src="https://user-images.githubusercontent.com/19422835/51401474-b52b3080-1aff-11e9-83c2-2fb1b19971f2.png">

**Comment by sdfgeoff** [comment]
This isn't exactly the same issue, but may have a similar fix. This blend has the bone animation exported incorrectly (and none of the bones have parents)
[tracks1.escn.blend.zip](https://github.com/godotengine/godot-blender-exporter/files/2954368/tracks1.escn.blend.zip)

**Comment by Jason0214** [comment]
@sdfgeoff finally got some time to test out this scene, however it works fine when export to godot3.0  
![out](https://user-images.githubusercontent.com/19422835/54857429-83c81000-4cbc-11e9-8011-722dd1513b96.gif)

when export to godot 3.1, the animation is broken but I can confirm it is the same issue as #142 (which I don't have much idea about how to fix..)

**Comment by doyouknowaboutq** [comment]
i wonder that does godot support non inherited bones? if godot does not support the non inherited bones, i mean rotation, scale or both, how plugin will handle the situation?

**Comment by Jason0214** [comment]
@doyouknowaboutq just bake the animation like baking IK

**Comment by doyouknowaboutq** [comment]
how can i that? if you mean "bake action" i did but there is no difference. still if body scales then head scales too. i think godot has no options for non inherital bones. all bones is inherited by default and only. i guess there is nothing to plugin do for now.

**Comment by Jason0214** [comment]
Yeah you cannot do it by just call `bake action` in blender. What I mean is that the idea is same as IK baking, the exporter would do some computation on each frame of non inherit bone. Instead of applying constraints in IK, here the exporter convert non inherit bones' transform from armature space to parent bone space.

**Comment by doyouknowaboutq** [comment]
@Jason0214 but there is a problem even if you baked the action

if you have a non-inherit scale/rotate head bone and a body bone which is connected with head bone and when you scale this body bone then animation works good in godot. body is scaling but head isn't scaling. this is good.

but if you rotate the head bone while body bone was scaling then animation corrupts in godot. head is scaling in the odd way.

**Comment by Jason0214** [comment]
@doyouknowaboutq  Sorry, I missed this thread :(

> but if you rotate the head bone while body bone was scaling then animation corrupts in godot. head is scaling in the odd way.
 
Yes, it can only keep the animation consistent between godot and blender.  if user want to further edit bones in godot, he still get bones chained together. 

> i solved this problem by cut off relationship of bones. they are child of root and head bone has copy location constrait to tail of body bone and blender and godot are equals with esnc exporter now.

this is a quite good idea, and I can see it is possible to implement it in this addon with some effort.
